src/common/java/net/minecraftforge/gradle/common/util/BaseRepo.java

```diff
@@ -124,12 +127,11 @@ public void attach(Project project) {
             File cache = Utils.getCache(project, "bundeled_repo");
             GradleRepositoryAdapter.add(project.getRepositories(), "BUNDELED_" + random, cache,
                     SimpleRepository.of(ArtifactProviderBuilder.begin(ArtifactIdentifier.class).provide(
-                            new ArtifactProvider<ArtifactIdentifier>() {
-                                @Override
-                                public Artifact getArtifact(ArtifactIdentifier artifact) {
-                                    return repos.stream().map(repo -> repo.getArtifact(artifact)).filter(Artifact::isPresent).findFirst().orElse(Artifact.none());
-                                }
-                            }
+                            artifact -> repos.stream()
+                                    .map(repo -> repo.getArtifact(artifact))
+                                    .filter(s -> s.isPresent())
```

Comment:
The fun thing is, I did change it to `.filter(Artifact::isPresent)`, but then javac complained that it was invalid somehow when I was loading it in via `includeBuild` and Gradle was building it for use as a plugin (see [`5561101` (#794)](https://github.com/MinecraftForge/ForgeGradle/pull/794/commits/5561101f15a09bce70359feb72f57fa5e31c918a) which reverted it). I'll have to re-investigate that, and if I can't find a solution then I'll add a comment explaining why it can't be a method reference (along with a comment suppression)

Reply:
Further checking showed that I had Java 16 as the JDK in my IDE, which is why rebuilding from the IDE showed no errors. Switching it to Java 8 and rebuilding now made it show the same error as compiling through Gradle ("formal and actual arguments list differ in length"). This makes me reasonably certain that this is either a Java 8 compiler bug, or a change made between Java 8 and 16.

I've added a comment explaining this, along with a comment inspection suppression to remove the warning from IntelliJ IDEA.